docs/pages/dsVideoDevice/ds-video-device_L3_TestProcedure.md

```diff
@@ -0,0 +1,368 @@
+        artifacts:
+          - "<path>/bin/hal_test"
+          - "<path>/bin/libut_control.so"
+          - "<path>/bin/Sink_4k_VideoDevice.yaml"
+          - "<path>/bin/run.sh"
+        execute:
+          - "chmod +x /opt/HAL/dsVideoDevice_L3/hal_test"
+          - "chmod +x /opt/HAL/dsVideoDevice_L3/run.sh"
+          - cp -rf /usr/lib/libdshal.so /opt/HAL/dsVideoDevice_L3/
+          - "ln -s /usr/lib/libds-hal.so /opt/HAL/dsVideoDevice_L3/libdshal.so"
+      test1_FrameratePrePostChangeCallBack_Verify:
+        <<: *defaults
+      test2_ZoomMode:
+        <<: *defaults
+        streams:
+          - "<URL Path>/streams/BounceBall_3840x2160_23.98fps.mp4"
+      test3_SetDisplayFramerate:
+        <<: *defaults
+        streams:
+          - "<URL Path>/streams/BounceBall_3840x2160_23.98fps.mp4"
+          - "<URL Path>/streams/BounceBall_3840x2160_24fps.mp4"
+      test4_FRFMode:
+        <<: *defaults
+        streams:
+          - "<URL Path>/streams/BounceBall_3840x2160_23.98fps.mp4"
+          - "<URL Path>/streams/BounceBall_3840x2160_24fps.mp4"
+
+```
+
+#### Test Suite Configuration
+
+Example Test Setup configuration File: `ut/host/tests/dsClasses/dsVideoDevice_test_suite.yml`
+
+Update the execute command according to the device path where `HAL` binaries are copied.
+
+```yaml
+dsVideoDevice:
+  description: "dsVideoDevice Device Settings testing profile"
+  test:
+    execute: "/tmp/run.sh -p /tmp/Sink_4k_VideoDevice.yaml"
+    type: UT-C  # Cunit tests (UT-C)
+
+```
+
+## Run Test Cases
+
+Once the environment is set up, you can execute the test cases with the following command
+
+```bash
+python <TestCaseName.py> --config </PATH>/ut/host/tests/configs/example_rack_config.yml --deviceConfig </PATH>/ut/host/tests/configs/deviceConfig.yml
+```
+
+## Streams Required
+
+Refer [ds-video-device_L3_Low-Level_TestSpecification.md](./ds-video-device_L3_Low-Level_TestSpecification.md#video-streams-requirement) for the stream details
+
+
+## Test Cases
+
+### dsVideoDevice_test1_FrameratePreChangePostChangeCallback_Verify.py
+
+#### Platform Supported - test01
+
+- Sink
+
+#### User Input Required - test01
+
+- No
+
+#### Acceptance Criteria - test01
+
+- Verify the display framerate change with pre and post change callbacks.
+
+#### Expected Results - test01
+
+- This test checks whether the callback is triggered before and after when the display framerate changes.
+
+#### Test Steps - test01
+
+- Run the test file: **`dsVideoDevice_test1_FrameratePreChangePostChangeCallback_Verify.py`**
+- Execution process:
+
+  Upon execution, the test will:
+
+  - Download all the required artifacts.
+  - Copy them to the target directory.
+  - Automatically start the test execution.
+
+- Iterating through supported display framerates:
+
+  The test will repeat for different supported display framerates. Before that set the FRF mode to `Disable`. And, then on setting each framerates callback need to be triggered accordingly.
```

Comment:
No where described about the Pre and Post change callbacks?

Reply:
updated